src/thread/n3ds/SDL_syssem.c

```diff
@@ -62,29 +62,23 @@ void SDL_DestroySemaphore(SDL_sem *sem)
     }
 }
 
-int SDL_SemTryWait(SDL_sem *sem)
-{
-    if (sem == NULL) {
-        return SDL_InvalidParamError("sem");
-    }
-
-    return SDL_SemWaitTimeout(sem, 0);
-}
-
-int SDL_SemWaitTimeout(SDL_sem *sem, Uint32 timeout)
+int SDL_SemWaitTimeoutNS(SDL_sem *sem, Sint64 timeoutNS)
 {
     int retval;
 
     if (sem == NULL) {
         return SDL_InvalidParamError("sem");
     }
 
-    if (timeout == SDL_MUTEX_MAXWAIT) {
+    if (timeoutNS == SDL_MUTEX_MAXWAIT) {
         LightSemaphore_Acquire(&sem->semaphore, 1);
         retval = 0;
     } else {
         int return_code = LightSemaphore_TryAcquire(&sem->semaphore, 1);
         if (return_code != 0) {
+            /* FIXME: Does this code guarantee a wall clock timeout here?
```

Comment:
Since there isn't a function to wait on a `LightSemaphore`, this was just a hack to simulate it.

The 3DS seems to allow for sub-millisecond delays, so we could have:
```c
Uint64 timeout = SDL_GetTicksNS() + timeout;
while (SDL_GetTicksNS() < timeout) {
  /* Try acquire semaphore */
  /* SDL_DelayNS(1)? */
}
```

In the best scenario, `SDL_DelayNS(1)` takes about 9-10µs.

Other than that, the rest works great!

Reply:
I wasn't sure if that short of a wait might starve whatever code would release the semaphore, but I can certainly change it to that if you want.